### PR TITLE
Automated cherry pick of #103019: Fix frameworkImpl.extenders being not set

### PR DESCRIPTION
**Pull request by ordovicia**
Cherry pick of #103019 on release-1.20.

#103019: Fix frameworkImpl.extenders being not set

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Fixed a regression in 1.20+ scheduler extender setup that scheduler extenders are not called on preemptions
```

### CODE REVIEW

**Comment by ahg-g** [comment]
/ok-to-test
/triage accepted
/kind bug
/priority important-soon
@kubernetes/release-managers

**Comment by ahg-g** [comment]
you will need to adjust the tests a bit.

**Comment by ordovicia** [comment]
Adjusted the test. `make test` on `pkg/scheduler` succeeds on my machine.
PTAL. Thank you.

**Comment by ordovicia** [comment]
Am I supposed to do something, or can I just wait for approval?

**Comment by ahg-g** [comment]
> Am I supposed to do something, or can I just wait for approval?

you don't need to do anything, someone from release managers will approve it.

**Comment by alculquicondor** [comment]
@kubernetes/release-managers

**Review by cpanato — APPROVED** [review]
from RelEng:
/lgtm

**Pull request merged**
